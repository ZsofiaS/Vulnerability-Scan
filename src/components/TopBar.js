import React from 'react';
import Account from './Account';
import Notifications from './Notifications';
import './TopBar.css';
import data from '../data/scan';

const TopBar = () => {
  return (
    <div className='topBar'>
      <span>Welcome {data.user.displayName}</span>
      <div className='rightMenu'>
        <Notifications
          notifications={data.user.notifications}/>
        <Account
          name={data.user.displayName}
          email={data.user.email}
        />
      </div>
    </div>
  )
}

export default TopBar;
