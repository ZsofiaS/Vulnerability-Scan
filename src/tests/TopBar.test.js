import React from 'react';
import { shallow, mount } from 'enzyme';
import TopBar from '../components/TopBar';
import toJson from 'enzyme-to-json';
import data from '../data/scan';

describe('renders TopBar component', () => {
  it('renders without crashing', () => {
    shallow(<TopBar />);
  });
  it('renders welcome message', () => {
    data.user.displayName = 'Joe';
    const wrapper = shallow(<TopBar />);
    const message = (<span>Welcome Joe</span>);
    expect(wrapper.contains(message)).toEqual(true);
  });
});
