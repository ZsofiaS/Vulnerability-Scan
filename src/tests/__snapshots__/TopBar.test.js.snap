// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot TopBar snapshot 1`] = `
<div
  className="topBar"
>
  <span>
    Welcome 
    Joe
  </span>
  <div
    className="rightMenu"
  >
    <Notifications
      notifications={
        Array [
          Object {
            "date": "2020-01-01T00:24:29.000Z",
            "id": 232,
            "message": "Scan Complete (Monthly Vulnerability Scan)",
            "read": false,
          },
          Object {
            "date": "2019-12-01T00:22:16.000Z",
            "id": 230,
            "message": "Scan Complete (Monthly Vulnerability Scan)",
            "read": true,
          },
        ]
      }
    />
    <Account
      email="jane.doe@nccgroup.com"
      name="Joe"
    />
  </div>
</div>
`;
